src/honeybot/api/main.py

```diff
@@ -245,49 +241,24 @@ def greet(self):
             self.send(commands.join_channel(channel))
 
     def pull(self):
-        while self.isListenOn:
+        while self.is_listen_on:
             try:
-                data = self.irc.recv(2048)
-                raw_msg = data.decode("UTF-8")
-                msg = raw_msg.strip("\n\r")
-                self.stay_alive(msg)
-                self.core_commands_parse(msg)
-                logger.info(msg)
-
-                if len(data) == 0:
-                    try:
-                        logger.critical(f"<must handle reconnection - {len(data)}==0>")
-                        sys.exit()
-                    except Exception as e:
-                        logger.info(e)
+                data = self.irc.recv(2048).decode("UTF-8", errors="replace")
+                for line in data.split("\r\n"):
+                    if line != "":
+                        self.core_commands_parse(line)
+
+            except KeyboardInterrupt:
+                self.is_listen_on = False
+                self.quit()
             except Exception as e:
-                logger.info(e)
-
-    """
-    ONGOING REQUIREMENT/S
-    """
```

Comment:
@AnastasiosZyngiridis Why did you remove this section?

Reply:
@Abdur-rahmaanJ Sorry for the mistake. As they where ONGOING REQUIREMENT/S i didn't touch them and forgot to put them back in after i was done editing the rest of the code.